core/src/main/java/hudson/model/UsageStatistics.java

```diff
@@ -212,7 +215,11 @@ public Permission getRequiredGlobalConfigPagePermission() {
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
         try {
             // for backward compatibility reasons, this configuration is stored in Jenkins
-            Jenkins.get().setNoUsageStatistics(json.has("usageStatisticsCollected") ? null : true);
+            if (DISABLED) {
+                Jenkins.get().setNoUsageStatistics(Boolean.TRUE);
+            } else {
```

Comment:
persist the disabled nature so that any users of the API get the correct state.

Reply:
Unsure about this change.

By default, usage stats submission is enabled (when the field is still `null`), so any callers of the API that ignore the `DISABLED` flag will get the "correct" value only once the global configuration has been saved.

Right now, it's always wrong to rely on the getter alone (depending on use case). This change may tempt callers into ignoring the edge case of "global config has not yet been submitted".

Reply:
> By default, usage stats submission is enabled (when the field is still null), so any callers of the API that ignore the DISABLED flag will get the "correct" value only once the global configuration has been saved.

this is true today, and remains so after this change. 

However if the user previously enabled this usage stats, saved the config and then restarted Jenkins with `-Dhudson.model.UsageStatistics.disabled=true` they (as of this change) have no way to change the value in the UI, so it would forever be persisted as `FALSE`.

in no way does this change that any caller of `Jenkins.isUsageStatsCollected` can get invalid info as the value is `null` until a config change.

Reply:
> this is true today, and remains so after this change.

Right, but it's now an edge case more likely to be ignored / callers unaware of.

Don't feel strongly, just seems like something increasing the likelihood of being used wrong (which admittedly was quite substantial even before).